### PR TITLE
friendschat: add option to clear recent channel list

### PR DESCRIPTION
**Pull request by Hydrox6**
Currently, there's no way to clear the recent chat list without config surgery, which is pretty terrible. This PR adds an option to do this to the Friends Chat tile, similar to `Inventory Tags`.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/friendschat/FriendsChatPlugin.java [hunk 1]
```diff
@@ -715,4 +734,39 @@ private void colorIgnoredPlayers(Color ignoreColor)
...
 	}
+
+	@Subscribe
+	public void onWidgetMenuOptionClicked(WidgetMenuOptionClicked event)
+	{
+		if ((event.getWidget() == WidgetInfo.FIXED_VIEWPORT_FRIENDS_CHAT_TAB
+			|| event.getWidget() == WidgetInfo.RESIZABLE_VIEWPORT_FRIENDS_CHAT_TAB
+			|| event.getWidget() == WidgetInfo.RESIZABLE_VIEWPORT_BOTTOM_LINE_FRIEND_CHAT_TAB)
+			&& CLEAR_RECENT_OPTION.equals(event.getMenuOption())
+			&& CLEAR_RECENT_TARGET.equals(Text.removeTags(event.getMenuTarget())))
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/friendschat/FriendsChatPlugin.java:745]: It doesn't look like tags are added to the menu target by MenuManager, why is `removeTags()` necessary?

Comment (Hydrox6) [reply runelite-client/src/main/java/net/runelite/client/plugins/friendschat/FriendsChatPlugin.java:745]: I was copying `Menu Entry Swapper`'s shift click stuff. Though that the colour that the target has was added to the text, but I'll remove it.

Comment (Hydrox6) [reply runelite-client/src/main/java/net/runelite/client/plugins/friendschat/FriendsChatPlugin.java:745]: So it turns out that `removeTags` is there for a good reason
image

File: runelite-api/src/main/java/net/runelite/api/widgets/WidgetInfo.java [hunk 2]
```diff
@@ -276,6 +276,7 @@
 	RESIZABLE_VIEWPORT_BOTTOM_LINE_STATS_ICON(WidgetID.RESIZABLE_VIEWPORT_BOTTOM_LINE_GROUP_ID, WidgetID.ResizableViewportBottomLine.SKILLS_ICON),
 	RESIZABLE_VIEWPORT_BOTTOM_LINE_MAGIC_ICON(WidgetID.RESIZABLE_VIEWPORT_BOTTOM_LINE_GROUP_ID, WidgetID.ResizableViewportBottomLine.MAGIC_ICON),
 	RESIZABLE_VIEWPORT_BOTTOM_LINE_FRIEND_ICON(WidgetID.RESIZABLE_VIEWPORT_BOTTOM_LINE_GROUP_ID, WidgetID.ResizableViewportBottomLine.FRIEND_ICON),
+	RESIZABLE_VIEWPORT_BOTTOM_LINE_FRIEND_CHAT_TAB(WidgetID.RESIZABLE_VIEWPORT_BOTTOM_LINE_GROUP_ID, WidgetID.ResizableViewportBottomLine.FC_TAB),
```
Comment (Nightfirecat) [inline runelite-api/src/main/java/net/runelite/api/widgets/WidgetInfo.java:279]: To match the naming of the other viewports
```suggestion
	RESIZABLE_VIEWPORT_BOTTOM_LINE_FRIENDS_CHAT_TAB(WidgetID.RESIZABLE_VIEWPORT_BOTTOM_LINE_GROUP_ID, WidgetID.ResizableViewportBottomLine.FC_TAB),
```

**Review by Nightfirecat — APPROVED** [review]
LGTM otherwise

File: runelite-api/src/main/java/net/runelite/api/widgets/WidgetInfo.java (hunk 2 above)
Comment (Hydrox6) [reply runelite-api/src/main/java/net/runelite/api/widgets/WidgetInfo.java:279]: I did it to match the existing `RESIZABLE_VIEWPORT_BOTTOM_LINE_FRIEND_CHAT_ICON`. It's misnamed, but at this point all the `BOTTOM_LINE` viewports are misnamed, given we now know the names are `RESIZABLE_CLASSIC` and `RESIZABLE_MODERN`

**Pull request closed without merging**